crates/ruff_cli/src/commands/add_noqa.rs

```diff
@@ -56,7 +57,12 @@ pub(crate) fn add_noqa(
                 .and_then(|parent| package_roots.get(parent))
                 .and_then(|package| *package);
             let settings = resolver.resolve(path, pyproject_config);
-            match add_noqa_to_path(path, package, source_type, settings) {
+            let Ok(LintSource(source_kind)) = LintSource::try_from_path(path, source_type) else {
+                // TODO(dhruvmanila): Display the error to the user.
+                error!("Failed to extract source from {}", path.display());
```

Comment:
Could we include the error in the message?

Reply:
The error type is an enum where the first variant is an `IO` error while other is a `Diagnostics`. I'm not sure how to transform the `Diagnostics` into a meaningful message without the use of `Printer`.

Reply:
I feel like the underlying issue here is that `Notebook::from_path` uses `Box<Diagnostics>` as its error type. Can we instead return an error there and convert to diagnostics as-needed? We can use a dedicated error enum to represent the IO error vs. the "you passed in something that isn't a notebook" error.

Reply:
I'm going to do this refactor quickly because it's bothering me :joy:

Reply:
See: https://github.com/astral-sh/ruff/pull/7035.

Reply:
> I'm going to do this refactor quickly because it's bothering me 😂

No worries, thanks! :)